optuna/pruners/successive_halving.py

```diff
@@ -103,9 +114,16 @@ def prune(self, study, trial):
 
         rung = _get_current_rung(trial)
         value = trial.intermediate_values[step]
-        trials = None
+        trials = None  # type: Optional[List[FrozenTrial]]
 
         while True:
+            if self._auto_min_resource:
```

Comment:
How about modifying this line as follows to reduce unnecessary `_estimate_in_resource()` calls once `_min_resource` is set?
```suggestion
            if self._auto_min_resource and self._min_resource is None:
```

Reply:
Furthermore, I think that the `_auto_min_resource` field can be removed from the class because it seems only `_min_resource` is needed to handle this condition.